napari/layers/_layer_actions.py

```diff
@@ -77,6 +77,27 @@ def _project(ll: LayerList, axis: int = 0, mode='max'):
     ll.append(new)
 
 
+def _convert_dtype(ll: LayerList, mode='int64'):
+    layer = ll.selection.active
+    if not layer:
+        return
+    if layer._type_string != 'labels':
+        raise NotImplementedError(
+            "Data type conversion only implemented for labels"
+        )
+
+    target_dtype = np.dtype(mode)
+    if (
+        layer.data.min() < np.iinfo(target_dtype).min
+        or layer.data.max() > np.iinfo(target_dtype).max
```

Comment:
```suggestion
        np.min(layer.data) < np.iinfo(target_dtype).min
        or np.max(layer.data) > np.iinfo(target_dtype).max
```

This will work with non-NumPy arrays including tensorstore arrays (where it currently fails)